packages/core/src/dialects/abstract/query.d.ts

```diff
@@ -256,55 +230,61 @@ export class AbstractQuery {
 
   /**
    * Checks if the query starts with 'call'
-   *
-   * @returns {boolean}
    */
   isCallQuery(): boolean;
 
   /**
-   * @param {string} sql
-   * @param {Function} debugContext
-   * @param {Array|object} parameters
+   * @param sql
+   * @param debugContext
+   * @param parameters
    * @protected
-   * @returns {Function} A function to call after the query was completed.
+   * @returns A function to call after the query was completed.
    */
-  protected _logQuery(sql: string, debugContext: ((msg: string) => any), parameters: unknown[]): () => void;
+  protected _logQuery(
+    sql: string,
+    debugContext: ((msg: string) => any),
+    parameters: unknown[] | Record<string, unknown>
+  ): () => void;
 
   /**
    * The function takes the result of the query execution and groups
    * the associated data by the callee.
    *
    * Example:
-   *   groupJoinData([
-   *     {
-   *       some: 'data',
-   *       id: 1,
-   *       association: { foo: 'bar', id: 1 }
-   *     }, {
-   *       some: 'data',
-   *       id: 1,
-   *       association: { foo: 'bar', id: 2 }
-   *     }, {
-   *       some: 'data',
-   *       id: 1,
-   *       association: { foo: 'bar', id: 3 }
-   *     }
-   *   ])
+   * ```ts
+   * groupJoinData([
+   *   {
+   *     some: 'data',
+   *     id: 1,
+   *     association: { foo: 'bar', id: 1 }
+   *   }, {
+   *     some: 'data',
+   *     id: 1,
+   *     association: { foo: 'bar', id: 2 }
+   *   }, {
+   *     some: 'data',
+   *     id: 1,
+   *     association: { foo: 'bar', id: 3 }
+   *   }
+   * ]);
+   * ```
    *
    * Result:
    *   Something like this:
```

Comment:
This gets rendered as 'Result: Something like this:', I would just call this 'Result' and make it a header